test/index.js

```diff
@@ -16,1680 +15,1339 @@ const internals = {};
+        }).to.not.throw();
+    });
+
+    it('fails if validateFunc is not a function', () => {
 
-            expect(() => {
+        const server = new Hapi.Server();
+        server.register(require('../'));
 
-                server.auth.strategy('default', 'cookie', true, { password: 'password-should-be-32-characters' });
-            }).to.not.throw();
+        expect(() => {
 
-            done();
-        });
+            server.auth.strategy('session', 'cookie', { validateFunc: 'not a function' });
+        }).to.throw(Error);
     });
 
-    it('passes with a password configured which is a Buffer', (done) => {
+    it('fails if keepAlive is configured but not ttl', () => {
 
         const server = new Hapi.Server();
-        server.connection();
-        server.register(require('../'), (err) => {
+        server.register(require('../'));
+
+        expect(() => {
+
+            server.auth.strategy('session', 'cookie', {
+                password: 'password-should-be-32-characters',
+                keepAlive: true
+            });
+        }).to.throw(Error);
+    });
 
-            expect(err).to.not.exist();
+    it('authenticates a request', async () => {
+
+        const server = new Hapi.Server();
+        server.register(require('../'));
 
-            expect(() => {
+        server.auth.strategy('session', 'cookie', {
+            password: 'password-should-be-32-characters',
+            ttl: 60 * 1000,
+            domain: 'example.com',
+            cookie: 'special',
+            clearInvalid: true,
+            validateFunc: function (request, session) {
 
-                server.auth.strategy('default', 'cookie', true, { password: new Buffer('foobar') });
-            }).to.not.throw();
+                const override = Hoek.clone(session);
+                override.something = 'new';
 
-            done();
+                return {
+                    valid: session.user === 'valid',
+                    credentials: override
+                };
+            }
         });
-    });
 
-    it('fails if validateFunc is not a function', (done) => {
+        server.auth.default('session');
 
-        const server = new Hapi.Server();
-        server.connection();
-        server.register(require('../'), (err) => {
+        Helpers.loginWithResourceEndpoint(server);
 
-            expect(err).to.not.exist();
+        const res = await server.inject('/login/valid');
 
-            expect(() => {
+        expect(res.result).to.equal('valid');
+        const header = res.headers['set-cookie'];
+        expect(header.length).to.equal(1);
+        expect(header[0]).to.contain('Max-Age=60');
+        const cookie = header[0].match(/(?:[^\x00-\x20\(\)<>@\,;\:\\"\/\[\]\?\=\{\}\x7F]+)\s*=\s*(?:([^\x00-\x20\"\,\;\\\x7F]*))/);
 
-                server.auth.strategy('default', 'cookie', true, { validateFunc: 'not a function' });
-            }).to.throw(Error);
+        /* eslint-disable hapi/no-shadow-relaxed */
```

Comment:
I guess this flag is no longer necessary

Reply:
yeah, if you need to turn off linting in your tests your tests are broken - it looks like you've fixed the error this was meant to circumvent anyway though